### PR TITLE
Eagerly evaluate inline requires in Jest

### PR DESCRIPTION
**Pull request by gaearon**
I inlined some requires in #7188 to fix the build size regression.
However this caused an issue with Jest due to it resetting module registry between tests.

This is a temporary fix to #7240.
It should be reverted as part of #7178.

Test plan:

1. Verify #7240 is fixed by following its repro instructions. You would need to build React and copy `build/modules/*` into its `node_modules/react/lib/`.

2. Verify a React example (e.g. `basic-click-counter`) in the repo runs with minified and unminified React.

3. Check that the minified build size has not increased.

Reviewers: @zpao @keyanzhang 

(Yes I know it’s super ugly. The upside is I know Jest a little better now!)

### CODE REVIEW

**Comment by gaearon** [comment]
@cpojer Does this look reasonable to you as a stopgap solution? (We have a better solution in #7178.)

**Comment by gaearon** [comment]
@keyanzhang Do you think we’ll get #7178 in this week? If yes, maybe we don’t need to bother with this fix.

**Comment by keyanzhang** [comment]
@gaearon I'll get that PR ready within a few days. Just to clarify -- we run these tests with the development build right?

**Comment by cpojer** [comment]
Yeah this seems reasonable. I'm sorry for the pain.

I'm however unsure how rollup would solve this as people will use the dev code (react/lib/*), not the production build when writing unit tests.

**Comment by gaearon** [comment]
Yes, we run tests with development CommonJS build. Rollup would solve this because we can move requires back to the top of the file (like Jest prefers) but it would still eliminate dead code in prod. Browserify is not that smart which is why I inlined them, but I won't have to after switching to Rollup.

**Comment by cpojer** [comment]
oh, yeah, that makes sense!

File: src/isomorphic/classic/types/checkReactTypeSpec.js [hunk 1]
```diff
@@ -17,6 +17,17 @@ var ReactPropTypesSecret = require('ReactPropTypesSecret');
 var invariant = require('invariant');
 var warning = require('warning');
 
+var ReactComponentTreeDevtool;
+
+if (process.env.NODE_ENV === 'test') {
```
Comment (zpao) [inline src/isomorphic/classic/types/checkReactTypeSpec.js]: I fully expect this to break internally unless we check `if (process && process.env && process.env.NODE_ENV = 'test')`.

Comment (gaearon) [inline src/isomorphic/classic/types/checkReactTypeSpec.js]: Aah, `__DEV__` “works” because it’s how it’s called internally! :steve_jobs_fireworks:

**Comment by gaearon** [comment]
Okay, this should work, right? This is getting super hacky. 😢

**Comment by gaearon** [comment]
@zpao Shall we do it?

**Comment by zpao** [comment]
Sure, but let's make sure this doesn't stick around for long.

**Pull request merged**